packages/dropdown/src/utils/useMergeRefs.tsx

```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react';
+
+export type ReactRef<T> = React.RefCallback<T> | React.MutableRefObject<T>;
+
+export function assignRef<T = any>(
+  ref: ReactRef<T> | null | undefined,
+  value: T,
+) {
+  if (ref == null) return;
+
+  if (typeof ref === 'function') {
+    ref(value);
+    return;
+  }
+
+  try {
+    ref.current = value;
+  } catch (error) {
+    throw new Error(`Cannot assign value '${value}' to ref '${ref}'`);
+  }
+}
+
+export function mergeRefs<T>(...refs: Array<ReactRef<T> | null | undefined>) {
+  return (node: T | null) => {
+    refs.forEach(ref => {
+      assignRef(ref, node);
+    });
+  };
+}
+
+export function useMergeRefs<T>(
+  ...refs: Array<ReactRef<T> | null | undefined>
+) {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  return useMemo(() => mergeRefs(...refs), refs);
+}
```

Comment:
I think you might've mentioned this on one of the walkthroughs, but why can't we use `useForwardedRef` from our hooks package?